hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -892,7 +875,7 @@
               separately.
             '''
       stage: V3
-      tests: []
+      tests: ["chip_sw_all_escalation_resets"]
```

Comment:
https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/dv/env/seq_lib/chip_sw_all_escalation_resets_vseq.sv;l=54?q=sw_all_es&ss=opentitan%2Fopentitan:hw%2F
I think right now all_escalation test only has rv_plic in `reg_we_check`. Do you think that is enough to cover all the bus integrity alert? Or shall we have another test to explicitly probe integrity error and make sure alert fires?

Reply:
Afaik, the alert leaves the register node via the same signal as the `reg_we_check`.

So if the register node is generated correctly by the regtool, the bus integrity error should work correctly by construction.

We test the register node template in many block-level tests, so I don't think there is much risk there.

But yes, the test would not be explicit, but rather rely on other tests that test shared parts of the register node template.

Reply:
@matutem WDYT?

Reply:
But plic does not have a block level DV... I think that is why we wanted to check here in chip level.

Reply:
We also do not have a good FPV flow for it.
We could - spin up a sanity DV testbench for plic just to check integrity errors, tl bus, and alert.
Or we can do it in chip level here.

I think at this point, maybe a small dv block level testbench is easier?

Reply:
From code inspection, alerts are triggered by either incoming tlul errors (via tlul_cmd_intg_chk err_o) or we checks. Is it possible to formally verify if either triggers the alert output will fire? That may be lower overhead than developing a full dv environment. I agree with @cindychip the all_escalation test is missing one of the alert sources.

Reply:
@matutem when you mention formally verify, do you think some assertion like this is enough?
https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/pinmux/fpv/vip/pinmux_assert_fpv.sv;l=716

Reply:
All we need to verify is the alert is triggered by "we" or "incoming tlul error". And could count on tlul_cmd_intg_chk correctness because it is shared, so tested extensively.

Reply:
okay! I can add an assertion for that.

Reply:
Understood - I think I was mainly trying to make the argument that since the `reg_top` is generated code, it may be enough to check that the fatal integrity error output of the `reg_top` is correctly connected to the fatal alert at the top level by triggering one of the two alert sources inside the `reg_top` (the write enable checker in this case).

How long do you think it will take to create a mini testbench?
Can we just use the testbench generator and add a few automated tests?

Reply:
yes - the mini testbench should be like 2 hours?

Reply:
Sorry I think my comment crossed @matutem's comment mid-flight ;).

I think we are aligned - it would be lower effort to just add an assertion to the pinmux FPV environment that checks that both conditions trigger the alert. The rest should then be covered by construction, because `tlul_cmd_intg_chk` is already tested quite extensively as @matutem noted above.